app/src/main/java/pl/droidevs/books/library/BookActivity.java

```diff
@@ -91,27 +124,171 @@ protected void onCreate(Bundle savedInstanceState) {
 
         if (savedInstanceState != null) {
             this.animationBundle = savedInstanceState.getBundle(BUNDLE_EXTRAS);
+            this.bookId = (BookId) savedInstanceState.getSerializable(EXTRAS_BOOK_ID);
         } else {
             this.animationBundle = getIntent().getBundleExtra(BUNDLE_EXTRAS);
+            this.bookId = (BookId) getIntent().getSerializableExtra(EXTRAS_BOOK_ID);
         }
 
         setupAnimations();
         setupViewModel();
+
+
+        appBarLayout.addOnOffsetChangedListener((appBarLayout, verticalOffset) -> {
+            int linesCount = tvExpandedTitle.getLineCount();
```

Comment:
tutaj się bardzo dużo dzieje, trzeba to jakoś podzielić albo nawet wyrzucić do jakiegoś helpera (tam też podzielić na mniejsze metody)

Reply:
Wiem, wiem 😉 
Ale nie mam czasu na razie tego zrobić. A chciałem to dodać przed końcem kursu 😅